DSharpPlus/Utilities.cs

```diff
@@ -289,6 +391,31 @@ public static bool Contains(this string str, params char[] characters)
             return false;
         }
 
+        /// <summary>
+        /// An async version of Regex.Replace
+        /// </summary>
+        /// <param name="regex">The Regex object</param>
+        /// <param name="input">The string to search for a match</param>
+        /// <param name="replacementFn">A custom method that examines each match and returns either the original matched string or a replacement string</param>
+        /// <returns></returns>
+        public static async Task<string> ReplaceAsync(this Regex regex, string input, Func<Match, Task<string>> replacementFn)
```

Comment:
Should it really be public?

Reply:
That's just a utility method, similar ones around here were marked public as well, but I really don't mind making it internal (or even private).